### PR TITLE
Created a unittests for switchmap/dashboard/table/events.py

### PR DESCRIPTION
**Pull request by MayankJha014**
This PR enhances the test coverage for the switchmap.dashboard.table.events module by:

- Adding comprehensive unit tests for _RawCol, EventsRow, EventTable, and the table function.

- Ensuring edge cases are handled effectively, including scenarios with no events or empty rows.

- Utilizing mocking techniques for external dependencies to maintain isolation and reliability in tests.

Issue : #205 

## Summary by CodeRabbit

- **Tests**
	- Added comprehensive unit tests for event table functionalities
	- Implemented test scenarios for valid events, empty events, and single event processing
	- Verified table generation and row content accuracy

### CODE REVIEW

**Comment by MayankJha014** [comment]
@palisadoes Please Review it

File: tests/switchmap_/dashboard/table/test_event.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        self.assertIsInstance(table, Table)
+        self.assertEqual(len(table.items), 1)
+
+    def test_table_function_no_events(self):
+        """Test table function with no events."""
+        events = []
+        result = table(events)
+        self.assertFalse(result)
+
+    def test_table_function_with_events(self):
```
Comment (palisadoes) [inline tests/switchmap_/dashboard/table/test_event.py:34]: 1. Test the table(events) with a list of EventMeta objects versus a mock. That is what is sent to the function.
2. Ensure you get a list of EventTable rows with specific values

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Test the table(events) with a list of EventMeta objects versus a mock. That is what is sent to the function.
2. Ensure you get a list of EventTable rows with specific values

**Comment by palisadoes** [comment]
1. Instead of using mocks, instantiate with the expected data so that you get expected results. Search the code base more carefully to see what real data is expected and its formatting.
2. This give false hope

**Comment by MayankJha014** [comment]
> 1. Instead of using mocks, instantiate with the expected data so that you get expected results. Search the code base more carefully to see what real data is expected and its formatting.
> 2. This give false hope

ok

**Comment by MayankJha014** [comment]
@coderabbitai full review

**Comment by MayankJha014** [comment]
@palisadoes Check it out sir is this ok

File: tests/switchmap_/dashboard/table/test_event.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+"""Unit tests for the event table functionalities.
+
+This module contains tests for the `table`, `EventTable`, and `EventsRow`
+classes from the `switchmap.dashboard.table.events` module.
+"""
+
```
Comment (palisadoes) [inline tests/switchmap_/dashboard/table/test_event.py:6]: You'll need to add the ROOT_PATH methodology so that the script can be run from any directory. It makes it easier for the developer. The aim is to ensure that the tests can be run from any directory. This means we get fewer requests asking for help.  Here is an example for the directory above.

https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/tests/switchmap_/dashboard/table/test_index.py

See whether it can be run from any directory under switchmap-ng as a test.

It looks OK otherwise

It looks OK otherwise

**Review by palisadoes — COMMENTED** [review]
You'll need to add the ROOT_PATH methodology so that the script can be run from any directory. It makes it easier for the developer. The aim is to ensure that the tests can be run from any directory. This means we get fewer requests asking for help.  Here is an example for the directory above.

https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/tests/switchmap_/dashboard/table/test_index.py

See whether it can be run from any directory under switchmap-ng as a test.

It looks OK otherwise

It looks OK otherwise

**Review by palisadoes — CHANGES_REQUESTED** [review]
You'll need to add the ROOT_PATH methodology so that the script can be run from any directory. It makes it easier for the developer. The aim is to ensure that the tests can be run from any directory. This means we get fewer requests asking for help.  Here is an example for the directory above.

https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/tests/switchmap_/dashboard/table/test_index.py

See whether it can be run from any directory under switchmap-ng as a test.

It looks OK otherwise

It looks OK otherwise

File: tests/switchmap_/dashboard/table/test_event.py (hunk 2 above)
Comment (MayankJha014) [reply tests/switchmap_/dashboard/table/test_event.py:6]: Sir I added it

**Comment by MayankJha014** [comment]
> You'll need to add the ROOT_PATH methodology so that the script can be run from any directory. It makes it easier for the developer. The aim is to ensure that the tests can be run from any directory. This means we get fewer requests asking for help. Here is an example for the directory above.
> 
> https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/tests/switchmap_/dashboard/table/test_index.py
> 
> See whether it can be run from any directory under switchmap-ng as a test.
> 
> It looks OK otherwise
> 
> It looks OK otherwise

Sir I added it and Can you merge it now.

**Review by palisadoes — APPROVED** [review]
